From fe1c00fb21ef5d2cb7915468c15a62185ee50a5f Mon Sep 17 00:00:00 2001
From: Kent Russell <kent.russell@amd.com>
Date: Wed, 4 Oct 2017 16:10:06 -0400
Subject: [PATCH 5675/5855] drm/amdkfd: Update wait_queue references

These changed for 4.13, so change wait_queue_t to wait_queue_entry_t,
wq.task_list to wq.head, and wait.task_list to wait.head.
[4.13]

Change-Id: I7056f4213eb22aa2c6cddc3e775642e1401cf3b1
Signed-off-by: Kent Russell <kent.russell@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_events.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_events.c b/drivers/gpu/drm/amd/amdkfd/kfd_events.c
index 512eb3e..835e045 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_events.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_events.c
@@ -36,10 +36,10 @@
 #include <linux/device.h>
 
 /*
- * Wrapper around wait_queue_t (wait queue entry)
+ * Wrapper around wait_queue_entry_t (wait queue entry)
  */
 struct kfd_event_waiter {
-	wait_queue_t wait;
+	wait_queue_entry_t wait;
 	struct kfd_event *event; /* Event to wait for */
 	bool activated;		 /* Becomes true when event is signaled */
 };
@@ -199,7 +199,7 @@ static void destroy_event(struct kfd_process *p, struct kfd_event *ev)
 	struct kfd_event_waiter *waiter;
 
 	/* Wake up pending waiters. They will return failure */
-	list_for_each_entry(waiter, &ev->wq.task_list, wait.task_list)
+	list_for_each_entry(waiter, &ev->wq.head, wait.entry)
 		waiter->event = NULL;
 	wake_up_all(&ev->wq);
 
@@ -336,7 +336,7 @@ static void set_event(struct kfd_event *ev)
 	/* Auto reset if the list is non-empty and we're waking someone. */
 	ev->signaled = !ev->auto_reset || !waitqueue_active(&ev->wq);
 
-	list_for_each_entry(waiter, &ev->wq.task_list, wait.task_list)
+	list_for_each_entry(waiter, &ev->wq.head, wait.entry)
 		waiter->activated = true;
 
 	wake_up_all(&ev->wq);
-- 
2.7.4

